lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -1287,8 +1287,22 @@ sub _serialize_tags_and_ratings
         if $opts->{tags} && $inc->{tags};
     $self->_serialize_user_tag_list($parent_node, $entity, $inc, $stash)
         if $opts->{user_tags} && $inc->{user_tags};
-    $self->_serialize_genre_list($parent_node, $entity, $inc, $stash)
-        if $opts->{genres} && $inc->{genres};
+    if ($opts->{genres} && $inc->{genres}) {
+        my @genre_tags = sort_by { $_->tag->name } @{$opts->{genres}};
+        my $genres = [map {$_->tag->genre} @genre_tags];
+        # Total is not used here for parity with tag-list
+        my $genre_hash = {items => $genres};
+        my %genre_counts = map { $_->tag->genre_id => $_->count } @genre_tags;
+        $self->_serialize_genre_list(
+            $parent_node,
+            $genre_hash,
+            $inc,
+            $stash,
+            1, # Consider this always toplevel
```

Comment:
Right now it doesn't look like `_serialize_genre` uses `$toplevel` at all; do you remember why this is needed?

It makes more sense to me to change the behavior when toplevel isn't set (if needed) than set the flag when it isn't true.  Though if we did have additional data to output, like annotations or linked entities, I wouldn't expect that to be outputted here anyway.

Reply:
That seems sensible, fair point. I changed it to not be toplevel then and to use !toplevel instead of skip_attributes.

Reply:
And yeah, `_serialize_genre` doesn't use `$toplevel` yet, but it will eventually, so I left this for consistency with others already.